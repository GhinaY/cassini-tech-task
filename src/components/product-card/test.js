import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import mockListingsData from '../../utils/test-utils/mock-listings-data';
import ProductCard from '.';

const mockListingDetails = mockListingsData[0];

function renderComponent(props) {
  return render(
    <MemoryRouter>
      <ProductCard {...props} />
    </MemoryRouter>
  );
};

describe('ProductCard', () => {

  test('renders correctly', () => {
    const { asFragment } = renderComponent(mockListingDetails);

    expect(asFragment()).toMatchSnapshot();
  });
});

// Definitely needs a lot more tests, including functional ones, 
// but in the interests of time I'll skip them for now.
