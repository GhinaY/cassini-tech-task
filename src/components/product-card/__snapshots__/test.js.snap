// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders correctly 1`] = `
<DocumentFragment>
  <a
    class="ProductCardContainer"
    href="/product/1"
  >
    <img
      alt="the product Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
      class="ProductImage"
      src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
    />
    <div>
      <span
        class="ProductCategory"
      >
        men's clothing
      </span>
      <span
        class="ProductTitle"
      >
        Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
      </span>
      <div
        class="ProductNumbers"
      >
        <span>
          ID: 1
        </span>
        <span>
          Â£109.95
        </span>
      </div>
      <p
        class="ProductDescription"
      >
        Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
      </p>
    </div>
  </a>
</DocumentFragment>
`;
